packages/viz/src/lib/Layer.ts

```diff
@@ -1,34 +1,41 @@
+import { MapboxLayer } from '@deck.gl/mapbox';
 import { MVTLayer } from '@deck.gl/geo-layers';
-import { Deck } from '@deck.gl/core';
 import { Source } from './sources/Source';
 import { CARTOSource } from './sources/CARTOSource';
 import { DOSource } from './sources/DOSource';
 import { DOLayer } from './deck/DOLayer';
 import { defaultStyles, StyleProperties, Style } from './style';
+import { Popup } from './popups/Popup';
+import { DeckInstance } from './basemap/create-map';
 
 export class Layer {
+  private _options: LayerOptions = {};
   private _source: Source;
   private _style?: Style;
 
   // Deck.gl Map instance
-  private _deckInstance: Deck | undefined;
+  private _deckInstance: DeckInstance | undefined;
 
   // Instance to the DeckLayer of the instance
   // It cannot be a reference to (import { Layer } from '@deck.gl/core') because
   // the typing of getPickinfo method is different from TileLayer and Layer are
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   private _deckLayer?: any;
 
-  public id: string;
-
   constructor(
     source: string | Source,
     style: Style | StyleProperties = {},
     options: LayerOptions = {}
   ) {
     this._source = buildSource(source);
     this._style = buildStyle(style);
-    this.id = options.id || `${this._source.id}-${Date.now()}`;
+
+    const defaultId = `${this._source.id}-${Date.now()}`;
+    Object.assign(this._options, options, { id: defaultId });
```

Comment:
The linter will warn you about this but we'd rather use the rest operator than Object.assign, so we can make something like:

```js
this._options = {
  id: `${this._source.id}-${Date.now()}`,
  ...options
}
```

Reply:
Make senses to me